minio/api.py

```diff
@@ -176,6 +179,16 @@ def __init__(self, endpoint, access_key=None,
         self._user_agent = _DEFAULT_USER_AGENT
         self._trace_output_stream = None
 
+        # Set credentials if possible
+
+        if credentials is not None:
+            self._credentials = credentials
+        else:
+            self._credentials = Credentials(
+                provider=Static(access_key, secret_key, session_token)
```

Comment:
I would recommend checking if at least access_key is not None / empty and creating a default chain provider which would propagate the IAM functionality more effectively to consumers of the api.

Reply:
Do you mean adding a chain provider instead of `Static()` credentials?
I was thinking something like this?
```
if credentials is not None:
    self._credentials = credentials
else:
    self._credentials = Credentials(
        provider= Chain(
            providers=[
                Static(access_key, secret_key, session_token),
                EnvAWS(),
                EnvMinio(),
                IamEc2MetaData(),
            ]
        )
    )
```

For checking if `access_key` is None/Empty the `Chain provider` will check this when trying to retrieve like so

```
def retrieve(self):
    for provider in self._providers:
        creds = provider.retrieve()
        if ((creds.access_key is None or creds.access_key == "") and
            (creds.secret_key is None or creds.secret_key == "")):
            continue
        self._current = provider
        return creds
    self._current = None
    raise NoCredentialsProvider()
```